system/Cache/Handlers/FileHandler.php

```diff
@@ -84,6 +85,31 @@ public function get(string $key)
 
 	//--------------------------------------------------------------------
 
+	/**
+	 * Get an item from the cache, or execute the given Closure and store the result.
+	 *
+	 * @param string  $key      Cache item name
+	 * @param integer $ttl      Time to live
+	 * @param Closure $callback Callback return value
+	 *
+	 * @return mixed
+	 */
+	public function remember(string $key, int $ttl, Closure $callback)
+	{
+		$value = $this->get($key);
+
+		if (! is_null($value))
+		{
+			return $value;
+		}
+
+		$this->save($key, $value = $callback(), $ttl);
+
+		return $value;
+	}
```

Comment:
Except for DummyHandler, it seems all cache handlers have the same implementation logic for this method. I do not see any differences. Perhaps we can extract `$this->remember` into an abstract parent `BaseHandler` which these classes would extend. Then DummyHandler would just override this method to return null.

```php
abstract class BaseHandler implements CacheInterface
{
	/**
	 * Get an item from the cache, or execute the given Closure and store the result.
	 *
	 * @param string  $key      Cache item name
	 * @param integer $ttl      Time to live
	 * @param Closure $callback Callback return value
	 *
	 * @return mixed
	 */
	public function remember(string $key, int $ttl, Closure $callback)
	{
		$value = $this->get($key);

		if (! is_null($value))
		{
			return $value;
		}

		$this->save($key, $value = $callback(), $ttl);

		return $value;
	}
}

class FileHandler extends BaseHandler
{
//...
}

class DummyHandler extends BaseHandler
{
    public function remember(string $key, int $ttl, Closure $callback)
    {
        return null;
    }
}
```

Reply:
Yes definitely consolidate the code to an abstract base class.

Reply:
Agree with @paulbalandan, i will refactor with `BaseHandler`.